### ISSUE TITLE
Qarray.eigenstates() returns transposed eigenvectors

### ISSUE THREAD

**Issue by GabrieleRolleri** [EQuS/jaxquantum#36]
The current implementation of eigenstates() returns eigenvectors in the same format as jax.numpy.linalg.eigh(), such that the ith eigenvector is stored at `[:, i]`. While this is not a problem in and of itself, eigenstates() returns a batched Qarray with the wrong batched axis, i.e. the ith batched state is not an eigenvector but rather we still need to extract it as `[:, i]`.

**Comment by GabrieleRolleri** [comment]
Fixed by 5f713c1
